include/arch/arm/armv/armv8-a/64/armv/vcpu.h

```diff
@@ -14,18 +14,18 @@
 #include <drivers/timer/arm_generic.h>
 
 /* Note that the HCR_DC for ARMv8 disables S1 translation if enabled */
-#ifdef CONFIG_DISABLE_WFI_WFE_TRAPS
 /* Trap SMC and override CPSR.AIF */
 #define HCR_COMMON ( HCR_VM | HCR_RW | HCR_AMO | HCR_IMO | HCR_FMO )
-#else
-/* Trap WFI/WFE/SMC and override CPSR.AIF */
-#define HCR_COMMON ( HCR_TWI | HCR_TWE | HCR_VM | HCR_RW | HCR_AMO | HCR_IMO | HCR_FMO )
-#endif
 
 /* Allow native tasks to run at EL0, but restrict access */
 #define HCR_NATIVE ( HCR_COMMON | HCR_TGE | HCR_TVM | HCR_TTLB | HCR_DC \
                    | HCR_TAC | HCR_SWIO |  HCR_TSC )
+
+#ifdef CONFIG_DISABLE_WFI_WFE_TRAPS
 #define HCR_VCPU   ( HCR_COMMON | HCR_TSC)
+#else
+#define HCR_VCPU   ( HCR_COMMON | HCR_TSC | HCR_TWFX)
```

Comment:
I would add `HCR_TSC` to `HCR_COMMON` and drop the `HCR_TWFX` define. That's also what ARMv7 is doing already.

Reply:
> drop the HCR_TWFX define

Not sure what you meant here, I pushed some changes, do you mind checking?

Reply:
The define is there to simplify `HCR_TWE | HCR_TWI`, but overall the extra indirection makes things less clear I think.

Otherwise the changes you made are what I meant.